### PR TITLE
Validate edges in  switch

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: scenario-api/src/main/scala/pl/touk/nussknacker/engine/build/GraphBuilder.scala [hunk 1]
```diff
@@ -65,11 +65,11 @@ trait GraphBuilder[R] {
     creator(BranchEnd(node.BranchEndData(BranchEndDefinition(branchId, joinId))))
 
   def switch(id: String, expression: Expression, exprVal: String, nexts: Case*): R =
-    creator(SwitchNode(Switch(id, expression, exprVal), nexts.toList, None))
+    creator(SwitchNode(Switch(id, Some(expression), Some(exprVal)), nexts.toList, None))
 
   def switch(id: String, expression: Expression, exprVal: String,
              defaultNext: SubsequentNode, nexts: Case*): R =
-    creator(SwitchNode(Switch(id, expression, exprVal), nexts.toList, Some(defaultNext)))
+    creator(SwitchNode(Switch(id, Some(expression), Some(exprVal)), nexts.toList, Some(defaultNext)))
```
Comment (arkadius) [inline scenario-api/src/main/scala/pl/touk/nussknacker/engine/build/GraphBuilder.scala:72]: Can u add deprecation notice about defaultNext to SwitchNode as well?

**Review by arkadius — COMMENTED** [review]
Some little changes added

**Comment by arkadius** [comment]
Do you know why snapshot changed?

File: scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala [hunk 2]
```diff
@@ -120,7 +120,15 @@ object node {
   case class Filter(id: String, expression: Expression, isDisabled: Option[Boolean] = None,
                     additionalFields: Option[UserDefinedAdditionalNodeFields] = None) extends NodeData with Disableable with RealNodeData with DeadEndingData
 
-  case class Switch(id: String, expression: Expression, exprVal: String, additionalFields: Option[UserDefinedAdditionalNodeFields] = None) extends NodeData with RealNodeData with DeadEndingData
+  object Switch {
+
+    def apply(id: String): Switch = Switch(id, None, None)
+
+    def apply(id: String, expression: Expression, exprVal: String): Switch = Switch(id, Some(expression), Some(exprVal))
```
[the remark this answers is not in the document]
Comment (mproch) [reply scenario-api/src/main/scala/pl/touk/nussknacker/engine/graph/node.scala:127]: :+1: Cleaned up a bit

File: interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala [hunk 3]
```diff
@@ -23,13 +26,15 @@ case class ValidationPerformed(errors: List[ProcessCompilationError],
...
 object NodeDataValidator {
 
+  case class OutgoingEdge(target: String, edgeType: Option[EdgeType])
+
   def validate(nodeData: NodeData, modelData: ModelData,
                validationContext: ValidationContext,
                branchContexts: Map[String, ValidationContext],
-               getFragment: String => Option[CanonicalProcess]
+               getFragment: String => Option[CanonicalProcess],
+               outgoingEdges: List[OutgoingEdge]
```
[the remark this answers is not in the document]
Comment (mproch) [reply interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeDataValidator.scala:37]: I'd rather wait to FE change, remove Option from request and not change here

**Comment by mproch** [comment]
The snapshot changed as expression disappeared from choice template and the validation failed as expression from template wsa invalid

**Review by arkadius — APPROVED** [review]

**Pull request merged**